### PR TITLE
Fixes tests

### LINKED ISSUE #43: Test suite failing
Some of the tests are failing when running:
`python setup.py test`

```
test/test_admin.py FFF....                                                                                                                                                                                                                      [ 46%]
test/test_connection.py ...F...                                                                                                                                                                                                                 [ 93%]
test/test_utils.py .          
```

### PR DESCRIPTION
**Pull request by scardena**
* Fixes #43  and #39 
* Refactors a few things
* Hopefully we can get to set up circle-ci so it can run tests on each PR.

### CODE REVIEW

File: test/test_admin.py [hunk 1]
```diff
@@ -188,30 +191,20 @@ def test_users(admin):
         'action': 'READ',
         'resource_type': 'user',
         'resource': ['username']
```
Comment (scardena) [inline test/test_admin.py:193]: by default when a user is created, it gets WR permission on itself. I don't know if in the past this was different, but currently it doesn't make much sense to add WRITE permission on it, as it's a default permission.

File: .gitignore [hunk 2]
```diff
@@ -102,3 +102,6 @@ venv.bak/
 
 # mypy
 .mypy_cache/
+
+# macos stuff
+.idea/
```
Comment (igable) [inline .gitignore:107]: Should also add here:
`.DS_Store`

**Review by igable — APPROVED** [review]
Looks good. Small request inline.

**Comment by buzztroll** [comment]
I still get 1 failure locally:

```(venv) ➜  pystardog python setup.py test
running pytest
running egg_info
writing pystardog.egg-info/PKG-INFO
writing dependency_links to pystardog.egg-info/dependency_links.txt
writing requirements to pystardog.egg-info/requires.txt
writing top-level names to pystardog.egg-info/top_level.txt
reading manifest file 'pystardog.egg-info/SOURCES.txt'
writing manifest file 'pystardog.egg-info/SOURCES.txt'
running build_ext
============================= test session starts ==============================
platform darwin -- Python 3.9.2, pytest-3.7.2, py-1.9.0, pluggy-0.13.1
rootdir: /Users/bresnaha/Dev/pystardog, inifile:
collected 15 items

test/test_admin.py F......                                               [ 46%]
test/test_connection.py ...s...                                          [ 93%]
test/test_utils.py .                                                     [100%]

=================================== FAILURES ===================================
________________________________ test_databases ________________________________

admin = <stardog.admin.Admin object at 0x110325160>

    def test_databases(admin):
        assert len(admin.databases()) == 0

        # create database
        db = admin.new_database('db', {
            'search.enabled': True,
            'spatial.enabled': True
        })

        assert len(admin.databases()) == 1
        assert db.name == 'db'
        assert db.get_options('search.enabled', 'spatial.enabled') == {
            'search.enabled': True,
            'spatial.enabled': True
        }

        # change options
        db.offline()
        db.set_options({'spatial.enabled': False})
        db.online()

        assert db.get_options('search.enabled', 'spatial.enabled') == {
            'search.enabled': True,
            'spatial.enabled': False
        }

        # optimize
        db.optimize()

        # repair
        db.offline()
>       db.repair()

test/test_admin.py:67:
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _
stardog/admin.py:442: in repair
    self.client.post(self.path + '/repair')
stardog/http/client.py:29: in post
    return self.__wrap(self.session.post(self.url + path, **kwargs))
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _

self = <stardog.http.client.Client object at 0x1103255e0>
request = <Response [405]>

    def __wrap(self, request):
        if not request.ok:
            try:
                msg = request.json()
            except ValueError:
                # sometimes errors come as strings
                msg = {'message': request.text}

>           raise exceptions.StardogException('[{}] {}: {}'.format(
                request.status_code, msg.get('code', ''), msg.get(
                    'message', '')))
E           stardog.exceptions.StardogException: [405] :

stardog/http/client.py:51: StardogException
=============== 1 failed, 13 passed, 1 skipped in 12.22 seconds ================```

**Comment by buzztroll** [comment]
Tests pass on stardog 7.6.1.

**Review by buzztroll — APPROVED** [review]

**Review by cullan — APPROVED** [review]

File: .circleci/config.yml [hunk 3]
```diff
@@ -0,0 +1,39 @@
...
+          MYSQL_USER: user
+          MYSQL_PASSWORD: passw0rd
+    steps:
+      - checkout
+      - run: python --version
+      - run:
+          name: Install dependencies
+          command: |
+            curl http://packages.stardog.com/stardog.gpg.pub | sudo apt-key add
+            echo "deb http://packages.stardog.com/deb/ stable main" | sudo tee -a /etc/apt/sources.list
```
Comment (rmyers) [inline .circleci/config.yml:18]: Why install stardog, and not just run this in docker? A simple docker-compose file will be easier to maintain than making sure you can install this properly.

Comment (scardena) [reply .circleci/config.yml:12]: I will work on this on a next PR

**Pull request merged**